rotkehlchen/chain/evm/decoding/interfaces.py

```diff
@@ -437,15 +440,16 @@ def __init__(
         self.save_data_to_cache_method = save_data_to_cache_method
         self.read_data_from_cache_method = read_data_from_cache_method
         self.chain_id = chain_id
-        if self.chain_id is None:
-            self.cache_data = self.read_data_from_cache_method()
-        else:
-            self.cache_data = self.read_data_from_cache_method(chain_id)
+        self.cache_data = default_cache
```

Comment:
1. If you do go with the constructor argument it should be `empty_cache` or `initial_cache`. You should also change the docstring above to explain what `initial_cache` does.
2. I am still torn. I don't see it getting read before the initial population. I don't understand why this is needed.